export_watch.go

```diff
@@ -0,0 +1,229 @@
+// Adapted from
+// https://github.com/hashicorp/envconsul/blob/master/runner.go
+
+package jsonconsul
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	dep "github.com/hashicorp/consul-template/dependency"
+	"github.com/hashicorp/consul-template/watch"
+	"github.com/hashicorp/consul/api"
+)
+
+func (c *JsonExport) RunWatcher() {
+	runner, err := NewRunner(c, false)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	runner.Start()
+}
+
+type Runner struct {
+	sync.RWMutex
+
+	// Prefix is the KeyPrefixDependency associated with this Runner.
+	Prefix *dep.StoreKeyPrefix
+
+	// ErrCh and DoneCh are channels where errors and finish notifications occur.
+	ErrCh  chan error
+	DoneCh chan struct{}
+
+	// ExitCh is a channel for parent processes to read exit status values from
+	// the child processes.
+	ExitCh chan int
+
+	// config is the Config that created this Runner. It is used internally to
+	// construct other objects and pass data.
+	config *JsonExport
+
+	// client is the consul/api client.
+	client *api.Client
+
+	// once indicates the runner should get data exactly one time and then stop.
+	once bool
+
+	// minTimer and maxTimer are used for quiescence.
+	minTimer, maxTimer <-chan time.Time
+
+	// outStream and errStream are the io.Writer streams where the runner will
+	// write information.
+	outStream, errStream io.Writer
+
+	// watcher is the watcher this runner is using.
+	watcher *watch.Watcher
+
+	// data is the latest representation of the data from Consul.
+	data map[string][]*dep.KeyPair
+
+	// killSignal is the signal to send to kill the process.
+	killSignal os.Signal
+}
+
+// NewRunner accepts a JsonExport, and boolean value for once mode.
+func NewRunner(config *JsonExport, once bool) (*Runner, error) {
+	var err error
+
+	log.Printf("[INFO] (runner) creating new runner (once: %v)", once)
+
+	runner := &Runner{
+		config: config,
+		once:   once,
+	}
+
+	s := strings.TrimPrefix(config.Prefix, "/")
+	runner.Prefix, err = dep.ParseStoreKeyPrefix(s)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := runner.init(); err != nil {
+		return nil, err
+	}
+
+	return runner, nil
+}
+
+// Start creates a new runner and begins watching dependencies and quiescence
+// timers. This is the main event loop and will block until finished.
+func (r *Runner) Start() {
+	var (
+		exitCh <-chan int
+	)
+
+	log.Printf("[INFO] (runner) starting")
```

Comment:
`go vet` catches these types of things – would recommend setting that up and adding to CI.